### PR TITLE
GEODE-6846: Clean up PureJavaMode class

### PR DESCRIPTION
**Pull request by alb3rtobr**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [n/a] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### Extra info
Clean up of PureJavaMode class and related code taking for granted Geode run always in that mode.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/SharedLibrary.java [hunk 1]
```diff
@@ -32,12 +27,6 @@
  */
 public class SharedLibrary {
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/SharedLibrary.java]: fix the class comment. It now has nothing to do with a shared library.

File: geode-core/src/main/java/org/apache/geode/internal/VersionDescription.java [hunk 2]
```diff
@@ -119,7 +119,7 @@ public String getProperty(String key) {
   }
 
   private String getNativeCodeVersion() {
-    return SmHelper.getNativeVersion();
+    return "native code unavailable";
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/VersionDescription.java]: Instead of always logging in the banner that "native code" is unavailable, I think it is time to remove this message. Since we don't have native code this message is misleading. It can make someone think they have configured things wrong. I think you should go to the one caller of this method and delete that message and then get rid of getNativeCodeVersion

File: geode-core/src/main/java/org/apache/geode/internal/SharedLibrary.java [hunk 3]
```diff
@@ -129,107 +118,6 @@
     }
   }
 
-  /**
-   * @return true if this process is 64bit
-   * @throws RuntimeException if sun.arch.data.model doesn't fit expectations
-   */
-  public static boolean is64Bit() {
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/SharedLibrary.java]: Don't delete `is64Bit()` -- this method still has a purpose in AddressableMemoryManager without the native library.

File: geode-core/src/main/java/org/apache/geode/internal/offheap/AddressableMemoryManager.java [hunk 4]
```diff
@@ -60,9 +59,9 @@ public static long allocate(int size) {
       if (err.getMessage() != null && !err.getMessage().isEmpty()) {
         msg += " Cause: " + err.getMessage();
       }
-      if (!SharedLibrary.is64Bit() && size >= (1024 * 1024 * 1024)) {
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/offheap/AddressableMemoryManager.java]: Revert this change. Keep the `!SharedLibrary.is64Bit() && `.

File: geode-core/src/main/java/org/apache/geode/internal/offheap/AddressableMemoryManager.java [hunk 5]
```diff
@@ -60,9 +59,9 @@ public static long allocate(int size) {
...
+      if (size >= (1024 * 1024 * 1024)) {
         msg +=
-            " The JVM looks like a 32-bit one. For large amounts of off-heap memory a 64-bit JVM is needed.";
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/offheap/AddressableMemoryManager.java]: Revert this change. Keep the original wording.

File: geode-core/src/main/java/org/apache/geode/internal/SharedLibrary.java (hunk 1 above)
Comment (alb3rtobr) [reply geode-core/src/main/java/org/apache/geode/internal/SharedLibrary.java]: Sounds fine. I was going to ask if this class should be set as deprecated, but renaming it sounds better.

File: geode-core/src/main/java/org/apache/geode/internal/PureJavaMode.java [hunk 6]
```diff
@@ -15,63 +15,13 @@
 
 package org.apache.geode.internal;
 
-import org.apache.geode.distributed.internal.DistributionConfig;
 import org.apache.geode.internal.statistics.OsStatisticsProvider;
 
 /**
  * Used to determine if product should use pure java mode.
  */
+@Deprecated
```
[the remark this answers is not in the document]
Comment (alb3rtobr) [reply geode-core/src/main/java/org/apache/geode/internal/PureJavaMode.java]: I cannot remove the class, it seems the `osStatsAreAvailable()` method is needed. I removed it in GEODE-6660 but @dschneider-pivotal said it was needed for some tool.

File: geode-core/src/main/java/org/apache/geode/internal/SharedLibrary.java (hunk 1 above)
Comment (alb3rtobr) [reply geode-core/src/main/java/org/apache/geode/internal/SharedLibrary.java]: Would you accept `JVMSizeUtils`? This gives more info about what "sizes" are the class about.

**Comment by dschneider-pivotal** [comment]
The org.apache.geode.management.internal.cli.commands.VersionCommandJUnitTest is now broken. It is looking for the native version string in the banner

**Comment by alb3rtobr** [comment]
> Just fix the broken test. I'm also okay with PureJavaMode being completely removed. We just need to cleanup the external dependencies before merging this PR.

Ok, so Im going to include the removal of the class then.

**Comment by alb3rtobr** [comment]
> Just fix the broken test. I'm also okay with PureJavaMode being completely removed. We just need to cleanup the external dependencies before merging this PR.

Ok, then I will remove the class
